src/main/java/com/pusher/client/connection/websocket/WebSocketConnection.java

```diff
@@ -303,6 +307,12 @@ public void run() {
         }, reconnectInterval, TimeUnit.SECONDS);
     }
 
+    // Received error codes 4000 >= 4099 indicate we shouldn't attempt reconnection
+    // https://pusher.com/docs/pusher_protocol#error-codes
+    private boolean shouldReconnect(int code) {
+        return !(code >= 4000 && code <= 4099);
```

Comment:
I'd personally find this easier to read without the negative, i.e. `code < 4000 || code >= 4100`, but it's not a big deal.